### ISSUE TITLE
When replaying a pipeline the code shown is the transformed declarative code rather than the original YAML

### ISSUE THREAD

**Issue by ferrucciobongianni** [jenkinsci/pipeline-as-yaml-plugin#41]
**Describe the bug**
I created a pipeline as YAML. Then ran it, so build n.1 completed. Clicked on the build, then hit the "replay" button, it showed the code of the pipeline, not as YAML anymore, but instead as its corresponding declarative code.

**To Reproduce**
Steps to reproduce the behaviour:
1. Create a simple "hello world" pipeline as YAML and save it.
2. Build the pipeline, wait until build completes.
3. Click on the build, then, click on "replay"
4. See the code is not YAML anymore. Instead, it is the declarative version of the previous YAML code.

**Expected behaviour**
Replaying a build from a pipeline defined as YAML should show the original YAML code.

**Example of YAML pipeline**
```
pipeline:
  agent:
    label: 'master'
  stages:
   - stage: Setup
     steps: echo "hello world"
```

**Declarative pipeline**
When replaying the code of the pipeline above, this is shown
```
pipeline {
  agent {
    node {
      label 'master'
    }
  }
  stages {
    stage('Setup') {
      steps {
        echo "hello world"
      }
    }
  }
}
```

**Comment by aytuncbeken** [comment]
Hi @ferrucciobongianni 

Replay option comes from another plugin that shows the script which is run. As this pipeline converts yaml format into declarative and passes converted script, behavior of the Replay is true. 

Another option can be to put "Replay with Yaml" link and show yaml content instead.

**Comment by ferrucciobongianni** [comment]
Hi @aytuncbeken,
Yes I realised that when I looked at the code. I agree the replay plugin behaves correctly, although I guess - from the user prospective - the expectation would be to see in the replay text box exactly what their input was (which, in this case, would be the YAML definition).
I think having a "replay with Yaml" would be ok. It might feel slightly inconsistent with the configuration of the pipeline, but still better than not having the option to modify the YAML code when replaying.

**Comment by aytuncbeken** [comment]
Agree, It is not the best option to have but at least something. I will work on this. Thanks for feedback!

**Comment by jonesbusy** [comment]
For the record: https://github.com/jenkinsci/workflow-cps-plugin/blob/master/plugin/src/main/java/org/jenkinsci/plugins/workflow/cps/replay/ReplayAction.java

I agreed the easiest would be to have a `ReplayYamlAction`
